### PR TITLE
rough draft of map

### PR DESCRIPTION
**Pull request by jordansread**
image

but the css isn't added automatically yet

### CODE REVIEW

File: scripts/visualize/visualize.matthew_map.R [hunk 1]
```diff
@@ -1,11 +1,13 @@
 visualize.matthew_map <- function(viz){
   
-  counties <- readData(viz[['depends']])
+  counties <- readData(viz[['depends']][1])
```
Comment (jordansread) [inline scripts/visualize/visualize.matthew_map.R:3]: what should we be doing for reading in multiple dependencies. what I did here ☝️ is fragile, but it works.

File: scripts/visualize/visualize.matthew_map.R [hunk 2]
```diff
@@ -25,9 +27,16 @@ visualize.matthew_map <- function(viz){
   
   for (i in 1:length(counties)){
     xml_attr(p[[i]], 'id') <- paste0('FIP-', as.character(counties$FIPS[i]))
+    xml_attr(p[[i]], 'class') <- 'county-polygon'
     xml_attr(p[[i]], 'style') <- NULL
     xml_attr(p[[i]], 'clip-path') <- NULL
   }
   
+  g.rivers <- xml_add_child(svg, 'g', id='rivers','class'='river-polyline')
```
Comment (jordansread) [inline scripts/visualize/visualize.matthew_map.R:35]: we don't ID rivers individually, so I grouped all of them into a single group id as `rivers` and class `river-polygon`

File: scripts/visualize/visualize.matthew_map.R [hunk 3]
```diff
@@ -25,9 +27,16 @@ visualize.matthew_map <- function(viz){
...
+  pl <- xml_find_all(svg, '//*[local-name()="polyline"]')
+  for (i in 1:length(pl)){
+    xml_add_child(g.rivers, 'polyline', points = xml_attr(pl[i], 'points'))
+  }
+  xml_remove(pl)
```
Comment (jordansread) [inline scripts/visualize/visualize.matthew_map.R:40]: in `XML` I can move child nodes around, but haven't figured out a good way to do that in `xml2`. Here I am just creating a new node for each and grabbing its geometry, then getting rid of the old nodes that weren't part of `rivers`

File: data/precip.json [hunk 4]
```diff
@@ -0,0 +1 @@
+[{"time":["2016-10-05 01:00:00"],"fips":["12005","12015","12035"],"values":[0,0.04,1]},{"time":["2016-10-05 02:00:00"],"fips":["12005","12015","12035"],"values":[0,0.08,1.3]}]
```
Comment (jordansread) [inline data/precip.json:1]: we are going to need to start messing around w/ the animation data soon...any suggestions for formatting @jiwalker-usgs ?

File: scripts/visualize/visualize.matthew_map.R (hunk 3 above)
Comment (wdwatkins) [reply scripts/visualize/visualize.matthew_map.R:40]: Sounds like you might be able to use the `.copy=FALSE` arg, haven't done it before though

File: scripts/visualize/visualize.matthew_map.R (hunk 1 above)
Comment (jiwalker-usgs) [reply scripts/visualize/visualize.matthew_map.R:3]: The depends section is really just to "build" the dependency graph, you should add a field where dependencies are actually told what to do.

File: scripts/visualize/visualize.matthew_map.R (hunk 2 above)
Comment (jiwalker-usgs) [reply scripts/visualize/visualize.matthew_map.R:35]: :+1:

File: viz.yaml [hunk 5]
```diff
@@ -69,7 +75,7 @@ visualize:
     id: matthew-map
     location: figures/matthew-water.svg
     visualizer: matthew_map
-    depends: "matthew-counties"
+    depends: ["matthew-counties","matthew-flowlines"]
     mapdata: "matthew-counties"
```
Comment (jiwalker-usgs) [inline viz.yaml:79]: you could add a `flowdata: matthew-flowlines` here

File: data/precip.json (hunk 4 above)
Comment (jiwalker-usgs) [reply data/precip.json:1]: I think we want to make fips into a key with value as the value.

**Pull request merged**